app/src/main/java/org/traccar/client/MainFragment.java

```diff
@@ -200,6 +214,36 @@ public boolean onOptionsItemSelected(MenuItem item) {
             startActivity(new Intent(getActivity(), AboutActivity.class));
             return true;
         }
+        // Aina 21-Jan-2019, Emergency alert on broadcast
+        else if (item.getItemId() == R.id.sosBroadcast) {
+            AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+            LayoutInflater inflater = getActivity().getLayoutInflater();
+            View dialogView = inflater.inflate(R.layout.sos_setting_field, null);
+            final EditText sosEditText = (EditText) dialogView.findViewById(R.id.sosEditText);
+            String broadcast = sharedPreferences.getString(MainFragment.KEY_BROADCAST, "");
+            if(broadcast.trim().length() > 0) {
+                sosEditText.setText(broadcast);
+            }
+            builder.setMessage(R.string.settings_sos_broadcast_summary)
+                    .setTitle(R.string.settings_sos_broadcast_title)
+                    .setView(dialogView);
+            builder.setPositiveButton("Save", new DialogInterface.OnClickListener() {
```

Comment:
Hardcoded text is not acceptable.

Reply:
I would recommend using standard Android ok/cancel text to avoid translations:

https://developer.android.com/reference/android/R.string